nexus/reconfigurator/execution/src/cockroachdb.rs

```diff
@@ -0,0 +1,32 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Ensures CockroachDB cluster settings are set
+
+use anyhow::Context;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db::DataStore;
+use nexus_types::deployment::Blueprint;
+use slog::info;
+
+pub(crate) async fn ensure_cluster_settings(
```

Comment:
Is the general case of TOCTOU problems why we store the external/internal DNS generations in the blueprint table, to prevent execution if the blueprint's expectation is different from reality? If so I can implement that pattern for the particular case of CockroachDB settings in general (we should probably avoid mucking about with `SET CLUSTER SETTING` if we aren't certain of the current state!).

Reply:
The internal/external DNS ones are a little unusual because they don't have to be accurate -- they just need to be no newer than reality at planning-time, which allows us to detect at execution-time if our plan is definitely out of date.  There's a comment explaining it:
https://github.com/oxidecomputer/omicron/blob/4a0d382f966e3433f44881d7c82b93ff863220e5/nexus/reconfigurator/execution/src/dns.rs#L154-L219

The specific mechanism varies across different execution steps.  Where the execution step makes a request to something we control, we've used a generation number (usually the blueprint one?) and have the server reject requests with older generations.  I think it just depends on what execution does and what tools we have available to avoid the TOCTOU race.

Reply:
Makes sense. I will probably store the cluster version setting in the table and check it during execution, although I need to think about how that will impact a future implementation when we eventually get to a release that has the new `cluster.auto_upgrade.enabled` setting.